src/app/tech-carbon-estimator/tech-carbon-estimator.component.spec.ts

```diff
@@ -90,17 +76,28 @@ describe('TechCarbonEstimatorComponent', () => {
     // The form component expects a full EstimatorValues object or undefined, but we only need to test that the service is called
     const formValue = undefined as unknown as EstimatorValues;
     component.handleFormSubmit(formValue);
+
     expect(estimationServiceStub.calculateCarbonEstimation).toHaveBeenCalledWith(formValue);
-    expect(component.showEstimation).toBeTrue();
   });
 
-  it('should hide estimation if form is reset', () => {
-    component.showEstimation = true;
+  it('should set the carbonEstimation when handleFormSubmit is called', () => {
+    component.carbonEstimation = null;
+    fixture.detectChanges();
+    spyOn(estimationServiceStub, 'calculateCarbonEstimation' as never).and.callThrough();
```

Comment:
Had to go and look up what `as never` was for. There's a more explicit alternative, which I don't know whether it's any clearer?

`spyOn<Partial<CarbonEstimationService>, any>(estimationServiceStub, 'calculateCarbonEstimation').and.callThrough();`

Reply:
Ah, good spot, I think that line is actually unnecessary. I think it should be sufficient to mock the injected dependency via `TestBed.configureTestingModule` and it's probably best to avoid spying on private methods when possible. I've deleted it.